src/SiteAlias/SiteAliasManager.php

```diff
@@ -252,13 +252,18 @@ protected function buildSelf(PreflightArgsInterface $preflightArgs, Environment
         // Create the 'self' alias record. Note that the self
         // record will be named '@self' if it is manually constructed
         // here, and will otherwise have the name of the
-        // alias or site specification used by the user.
-        return new AliasRecord(
-            [
-                'root' => $root,
-                'uri' => $preflightArgs->uri(),
-            ],
-            '@self'
-        );
+        // alias or site specification used by the user. Also note that if we
+        // pass in a falsy uri the drush config (i.e drush.yml) can not override
+        // it.
+        $uri = $preflightArgs->uri();
+        $data = [
+            'root' => $root,
+            'uri' => $uri,
+        ];
+        if (!$uri) {
+            unset($data['uri']);
```

Comment:
Another minor style point: I'd rather see 'uri' set conditionally than unset conditionally. If there were a number of such elements, then I might suggest `$data = array_filter($data);` might be better, but in this case I'd favor `if ($uri) { $data['uri'] = $uri; }`